tools/restatectl/src/util.rs

```diff
@@ -151,3 +167,92 @@ pub enum RangeParamError {
     #[error(transparent)]
     ParseError(#[from] ParseIntError),
 }
+
+pub async fn read_modify_write<F, T, E>(
```

Comment:
You might get this for free if you implement `MetadataStore` for `MetadataProxySvcClient` because then you could use `MetadataStoreClient::read_modify_write`.

Reply:
Really good suggestion! Thank you